node_modules/col-rest/lib/rest/lti.js

```diff
@@ -103,7 +112,23 @@ module.exports = function(client) {
    * @param  {Response}       callback.response               The response object as returned by requestjs
    * @see col-lti/lib/rest.js for more information
    */
-  client.lti.whiteboardsLaunch = function(data, callback) {
-    client.request('/lti/whiteboards', 'POST', data, null, callback);
+  client.lti.whiteboardsLaunch = function(course, data, callback) {
+    var headers = {
+      'referer': getRefererHeader(course, 'whiteboards')
+    };
+    client.request('/lti/whiteboards', 'POST', data, headers, callback);
+  };
+
+  /**
+   * Get the referer header for a given course and tool
+   *
+   * @param  {Course}   course    The course in which the user is launching the LTI tool
+   * @param  {String}   tool      The tool that is being launched (One of `assetlibrary`, `whiteboards` or `engagementIndex`)
+   * @return {String}             A URL that can be used as the referer header for the launch of one of the LTI tools
+   * @api private
+   */
+  var getRefererHeader = function(course, tool) {
+    var protocol = (course.canvas.use_https) ? 'https' : 'http';
+    return protocol + '://' + course.canvas.canvas_api_domain + '/external_tools/' + tool;
```

Comment:
This header is only used during the tests. When you launch an LTI tool, LTI doesn't actually give you the parent URL of where that tool is being loaded from (which is why we look at the referer in the first place). This URL is missing the `/course/<id>` portion, which I'll add in the follow-up.

It doesn't really matter what this URL is for the tests, as long as it's different from the whiteboard tool URL. 

`protocol + '://' + course.canvas.canvas_api_domain + '/courses/' + course.id + '/external_tools/' + tool;`